### ISSUE TITLE
CUDNN_STATUS_INTERNAL_ERROR when training decoder on Cityscapes

### ISSUE THREAD

**Issue by danny2646x** [sacmehta/ESPNet#67]
Hey @sacmehta. Really awesome repository. It was really helpful in my research project on segmentation.

I faced the above CUDNN_STATUS_INERNAL_ERROR Runtime error while training the decoder.

This is what the error looks like.....

Traceback (most recent call last):
  File "main.py", line 422, in <module>
    trainValidateSegmentation(parser.parse_args())
  File "main.py", line 352, in trainValidateSegmentation
    lossTr, overall_acc_tr, per_class_acc_tr, per_class_iu_tr, mIOU_tr = train(args, trainLoader, model, criteria, optimizer, epoch)
  File "main.py", line 99, in train
    output = model(input_var)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/ubuntu/ESPNet-master/train/Model.py", line 384, in forward
    classifier = self.classifier(concat_features)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/module.py", line 491, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/ubuntu/anaconda3/envs/pytorch_p36/lib/python3.6/site-packages/torch/nn/modules/conv.py", line 796, in forward
    output_padding, self.groups, self.dilation)
RuntimeError: cuDNN error: ### CUDNN_STATUS_INTERNAL_ERROR

Let me tell you the steps I followed before I started training the decoder - 

1. Annotated the cityscapes dataset.
2. Successfully trained the encoder.
3. Passed the encoder with the highest mIOU as the pretrained weights to train the decoder.

This is the command I am using to train the decoder - 

python3 main.py --batch_size 2 --scaleIn 1 --decoder True --pretrained results_enc__enc_2_8/model_20.pth --data_dir /cityscapes --logFile trainValLogdecoder.txt

However, this error cropped up everytime I started training the decoder.

I tried to debug this and I found out the following - 

Each time the Transform is scaling the dataset by (1024, 512), it shows this error. Other shapes like (1280, 720), (1280, 640), (512, 256) all worked smoothly. Only (1024, 512) is giving this error.

My Python Virtual Environment  Library Versions- 

CUDA - 10.0
PyTorch - 1.0
cuDNN - 7.3.1

I use a Tesla GPU for deep learning.

Can you please tell me what is the reason behind this? And how to resolve it?

**Comment by sacmehta** [comment]
Interesting. 512x256 is working, so technically 1024x512 should work.

Could you try disabling cuDNN and see if it works? Training will be slow, but just check to ensure that it is not a cuDNN error

**Comment by danny2646x** [comment]
Hello @sacmehta. Thanks for your early reply.

I followed your suggestion and I tried to use my CPU for training. I passed the onGPU argument to the script with the value False and moved all the tensors to the CPU accordingly and ran it.

And I waited.

Okay, after an agonisingly long wait the first epoch completed without any errors. So I guess there is an issue with the Nvidida Tesla GPU but again I am not sure why this CUDNN_INTERNSL_STATUS_ERROR should arise with a particular size (1024, 512). 

I thought there might be an issue with the GPU cache and so I cleared it (rm -rf ~/.nv ) and ran again on the GPU.  This didn't work.

Another angle I approached this issue was reducing the batch size in case this is all because of the larger image size making the GPU run out of memory. So I reduced the batch size to 4, then 2 and tried again. This didn't work as well. Interestingly, though, the size (1536, 786) being larger than (1024, 512), with a batch size of 6 did work like I said before.

Note - I have never faced this error before throughout the time I spent researching in deep learning. My GPU has never failed. So, I don't think this is probably a case of bad configuration.

@sacmehta , is there anything I can do? Or is training the decoder on any dataset with arbitrary sizes would suffice to have a good segmentation model?
